internal/cmd/controller/gitjob/gitjob.go

```diff
@@ -305,19 +319,300 @@ func (r *GitJobReconciler) newJob(ctx context.Context, obj *v1.GitJob) (*batchv1
 				Name:  "COMMIT",
 				Value: obj.Status.Commit,
 			},
-			corev1.EnvVar{
-				Name:  "EVENT_TYPE",
-				Value: obj.Status.Event,
-			},
 		)
 		job.Spec.Template.Spec.Containers[i].Env = append(job.Spec.Template.Spec.Containers[i].Env, proxyEnvVars()...)
 	}
 
 	return job, nil
 }
 
-func (r *GitJobReconciler) generateInitContainer(ctx context.Context, obj *v1.GitJob) (corev1.Container, error) {
-	args := []string{obj.Spec.Git.Repo, "/workspace"}
+func (r *GitJobReconciler) computeJobSpec(ctx context.Context, gitrepo *v1alpha1.GitRepo) (*batchv1.JobSpec, error) {
+	paths := gitrepo.Spec.Paths
+	if len(paths) == 0 {
+		paths = []string{"."}
+	}
+
+	// XXX: how much sense does it make to recompute the whole configmap here just to extract its name? Can we do
+	// better/simpler?
```

Comment:
```suggestion
	// calculate configmap content, because the name contains a hash
```

Yes, this seems weird. The config map is created by the other controller and the name contains a hash over the data.
Maybe we can fix to something fixed, like `gitrepo.name + gitrepo.commit`, but the current approach seems more fault tolerant, i.e. target configmaps from older/newer gitrepos would still work if the data didn't change?

Maybe we just document the current situation and move on..

Reply:
Agreed, documentation is a good approach for now. Thanks!